File: DockerLogDriver/src/main/java/de/derteufelqwe/driver/messages/VolumeDriver.java
package de.derteufelqwe.driver.messages;

import com.google.gson.annotations.SerializedName;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.io.Serializable;
import java.util.Map;

public class VolumeDriver {

    @Data
    @AllArgsConstructor
    public static class RCapabilities implements Serializable {

    }

    @Data
    @NoArgsConstructor
    @AllArgsConstructor
    public static class Capabilities implements Serializable {

        @SerializedName("Capabilities")
        private Caps capabilities = new Caps();


        @Data
        @NoArgsConstructor
        @AllArgsConstructor
        private static class Caps implements Serializable {

            @SerializedName("Scope")
            private String scope = "local";

        }

    }


    @Data
    @NoArgsConstructor
    @AllArgsConstructor
    public static class RGet implements Serializable {

        @SerializedName("Name")
        private String volumeName;

    }

    @Data
    @NoArgsConstructor
    @AllArgsConstructor
    public static class Get implements Serializable {

        @SerializedName("Volume")
        private Volume volume;

        @SerializedName("Err")
        private String error;

    }


    @Data
    @NoArgsConstructor
    @AllArgsConstructor
    public static class RCreate implements Serializable {

        @SerializedName("Name")
        private String name;

        @SerializedName("Opts")
        private Map<String, String> opts;

    }

    @Data
    @NoArgsConstructor
    @AllArgsConstructor
    public static class Create implements Serializable {

        @SerializedName("Err")
        private String error;

    }


    @Data
    @NoArgsConstructor
    @AllArgsConstructor
    public static class RRemove implements Serializable {

        @SerializedName("Name")
        private String volumeName;

    }

    @Data
    @NoArgsConstructor
    @AllArgsConstructor
    public static class Remove implements Serializable {

        @SerializedName("Err")
        private String error;

    }


    @Data
    @NoArgsConstructor
    @AllArgsConstructor
    public static class RMount implements Serializable {

        @SerializedName("Name")
        private String volumeName;

        @SerializedName("ID")
        private String id;

    }

    @Data
    @NoArgsConstructor
    @AllArgsConstructor
    public static class Mount implements Serializable {

        @SerializedName("Mountpoint")
        private String mountpoint;

        @SerializedName("Err")
        private String error;

    }


    @Data
    @NoArgsConstructor
    @AllArgsConstructor
    public static class RUnmount implements Serializable {

        @SerializedName("Name")
        private String volumeName;

        @SerializedName("ID")
        private String id;

    }

    @Data
    @NoArgsConstructor
    @AllArgsConstructor
    public static class Unmount implements Serializable {

        @SerializedName("Err")
        private String error;

    }


    @Data
    @NoArgsConstructor
    @AllArgsConstructor
    public static class RPath implements Serializable {

        @SerializedName("Name")
        private String volumeName;

    }

    @Data
    @NoArgsConstructor
    @AllArgsConstructor
    public static class Path implements Serializable {

        @SerializedName("Mountpoint")
        private String mountpoint;

        @SerializedName("Err")
        private String error;

    }


    @Data
    @AllArgsConstructor
    public static class RList implements Serializable {

    }

    @Data
    @NoArgsConstructor
    @AllArgsConstructor
    public static class List implements Serializable {

        @SerializedName("Volumes")
        private java.util.List<Volume> volumes;

        @SerializedName("Err")
        private String error;

    }


    // -----  Data classes  -----

    @Data
    @NoArgsConstructor
    @AllArgsConstructor
    public static class Volume implements Serializable {

        @SerializedName("Name")
        private String name;

        @SerializedName("Mountpoint")
        private String mountPoint;

        @SerializedName("Status")
        private Map<String, String> status;

    }

}

File: DockerPlugin/src/main/java/de/derteufelqwe/plugin/messages/VolumeDriver.kt
package de.derteufelqwe.plugin.messages

import com.google.gson.annotations.SerializedName
import java.io.Serializable

class VolumeDriver {

    class RCapabilities : Serializable

    data class Capabilities(
        @SerializedName("Capabilities") var capabilities: Caps = Caps(),
    ) : Serializable {

        data class Caps(
            @SerializedName("Scope") var scope: String = "global"
        ) : Serializable
    }


    data class RGet(
        @SerializedName("Name") var volumeName: String? = null
    ) : Serializable

    data class Get(
        @SerializedName("Volume") var volume: Volume? = null,
        @SerializedName("Err") var error: String? = null
    ) : Serializable


    class RCreate(
        @SerializedName("Name") var name: String? = null,
        @SerializedName("Opts") var opts: Map<String, String> = HashMap()
    ) : Serializable

    data class Create(
        @SerializedName("Err") var error: String? = null
    ) : Serializable


    data class RRemove(
        @SerializedName("Name") var volumeName: String? = null
    ) : Serializable

    data class Remove(
        @SerializedName("Err") var error: String? = null
    ) : Serializable


    data class RMount(
        @SerializedName("Name") var volumeName: String? = null,
        @SerializedName("ID") var id: String? = null
    ) : Serializable

    data class Mount(
        @SerializedName("Mountpoint") var mountpoint: String? = null,
        @SerializedName("Err") var error: String? = null
    ) : Serializable


    data class RUnmount(
        @SerializedName("Name") var volumeName: String? = null,
        @SerializedName("ID") var id: String? = null
    ) : Serializable

    data class Unmount(
        @SerializedName("Err") var error: String? = null
    ) : Serializable


    data class RPath(
        @SerializedName("Name") var volumeName: String? = null
    ) : Serializable

    data class Path(
        @SerializedName("Mountpoint") var mountpoint: String? = null,
        @SerializedName("Err") var error: String? = null
    ) : Serializable


    class RList : Serializable

    data class List(
        @SerializedName("Volumes") var volumes: kotlin.collections.List<Volume>? = null,
        @SerializedName("Err") var error: String? = null
    ) : Serializable


    // -----  Sub classes  -----


    data class Volume(
        @SerializedName("Name") var name: String? = null,
        @SerializedName("Mountpoint") var mountPoint: String? = null,
        @SerializedName("Status") var status: Map<String, String>? = null,
        @SerializedName("CreatedAt") var created: String? = null
    ) : Serializable
}
